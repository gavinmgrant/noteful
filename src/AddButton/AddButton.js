import React from 'react'
import './AddButton.css'

export default function AddButton(props) {
  const { tag, className, children, ...otherProps } = props

  return React.createElement(
    props.tag,
    {
      className: ['AddButton', props.className].join(' '),
      ...otherProps
    },
    props.children
  )
}

AddButton.defaultProps ={
  tag: 'a',
}