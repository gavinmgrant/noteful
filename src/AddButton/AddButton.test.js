import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddButton from './AddButton'

describe(`AddButton component`, () => {
    const props = {
        tag: 'a',
        className: 'test-class-name',
        children: <p>test children</p>,
        'data-other': 'test-other-prop'
    }

    it('renders a button.AddButton by default', () => {
        const wrapper = shallow(<AddButton />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders the add button from props', () => {
        const wrapper = shallow(<AddButton {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})