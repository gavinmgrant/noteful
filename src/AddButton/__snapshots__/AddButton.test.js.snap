// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton component renders a button.AddButton by default 1`] = `
<a
  className="AddButton "
/>
`;

exports[`AddButton component renders the add button from props 1`] = `
<a
  className="AddButton test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</a>
`;
